### ISSUE TITLE
Bug: sam cli with Terraform.  sam local invoke MemorySize is always 128MB even though I change it in terraform resource.

### ISSUE THREAD

**Issue by dedosmedia** [aws/aws-sam-cli#4824]
### Description:
I am following the article: https://aws.amazon.com/es/blogs/compute/better-together-aws-sam-cli-and-hashicorp-terraform/
running any of the samples provided.

### Steps to reproduce:
```bash
git clone https://github.com/aws-samples/aws-sam-terraform-examples
cd aws-sam-terraform-examples/zip_based_lambda_functions/lambda-example
```

Edit main.tf and add a memory_size to the lambda function resource
```terraform
resource "aws_lambda_function" "hello-terraform" {
     ...
    memory_size = 1024
     ...
}
```
```bash
terraform init
terraform apply
sam build --hook-name terraform --beta-features
sam local invoke  --beta-features 
```

### Observed result:
The lambda logs report:
```
END RequestId: 09b77c0c-e5b7-4913-b1d5-5b427c7f1bfd
REPORT RequestId: 09b77c0c-e5b7-4913-b1d5-5b427c7f1bfd  Init Duration: 0.03 ms  Duration: 111.26 ms     Billed Duration: 112 ms
Memory Size: 128 MB     Max Memory Used: 128 MB
```
Notice the reported Memory Size for the function remains as the 128MB default value

### Expected result:

```
END RequestId: 09b77c0c-e5b7-4913-b1d5-5b427c7f1bfd
REPORT RequestId: 09b77c0c-e5b7-4913-b1d5-5b427c7f1bfd  Init Duration: 0.03 ms  Duration: 111.26 ms     Billed Duration: 112 ms
Memory Size: 1024 MB     Max Memory Used: 128 MB
```
It should report the memory_size configured in terraform resource block, 1024MB in this case. It's killing my function because the max memory is more than 128MB even though my desired Memory Size is 1024MB. 

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

**OS:** 
Linux Mint 5.15.0-56-generic #62-Ubuntu SMP Tue Nov 22 19:54:14 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

**sam --info**
```json
{
  "version": "1.76.0",
  "system": {
    "python": "3.7.10",
    "os": "Linux-5.15.0-56-generic-x86_64-with-debian-bookworm-sid"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.22",
    "aws_cdk": "Not available",
    "terraform": "1.3.2"
  }
}
```

### workaround
Before running sam local invoke, add manually the MemorySize: 1024 field into the .aws-sam/build/template.yaml generated by sam build. This way the lambda logs show the right memory size and my function is not killed when more than 128MB is being used.
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  ModuleAwsLambdaAwsLambdaFunctionThis04E96302B:
    Type: AWS::Lambda::Function
    Properties:
      FunctionName: blender-dev-render
      ...
      MemorySize: 1024
      ...
```

**Comment by mndeveci** [comment]
Thanks for raising this issue and deep dive @dedosmedia !

If this is fixed by the change that you provided, we would appreciate if you could raise a pull request for it!

**Comment by moelasmar** [comment]
The fix of this issue was released in the new SAM CLI release 1.77.0
